### PR TITLE
fix when the user comes back while loading the kid

### PR DESCRIPTION
**Pull request by madclem**
Problem: While the kid is login in (during the loader), if the user presses the back button the webview still shows after while (because once the kid is logged in, it updates the `auth/index` props and shows the <Game /> component. 

There is now an extra check, see if we still want to load the kid (add a state variable to `auto/index` with the kid we want to load, if any)

### CODE REVIEW

File: src/navigation/auth/index.js [hunk 1]
```diff
@@ -24,20 +29,62 @@ class Auth extends Component {
...
+  onKidSelect = async (kid) => {
+    // gets called from the <Home /> component, when the user selects a kid to login
+    if (this.props.loggedIn) { // maybe uncessary: if already logged in (asynchronously): disconnect the current kid
+      this.props.authLogoutKid()
+    }
+    this.setState({ // update the state to say we're trying to log in with this specific kid
+      selectedKid: kid
+    })
+  }
+
```
Comment (madclem) [inline src/navigation/auth/index.js]: when selecting a kid, update the state to say we're actually trying to log in as a kid

File: src/navigation/auth/index.js [hunk 2]
```diff
@@ -24,20 +29,62 @@ class Auth extends Component {
...
+  onBackPressed = () => {
+    if (this.state.selectedKid) {
+      // when back is pressed, if you were trying to load as a kid, set the state to say you're not trying anymore (back on the home)
+      this.setState({
+        selectedKid: null
+      })
+    }
+  }
+
```
Comment (madclem) [inline src/navigation/auth/index.js]: if we were trying to log in as a kid and back is pressed, we set to null the selectedKid variable in the state, it's clear we do not want anymore!

File: src/navigation/auth/index.js [hunk 3]
```diff
@@ -24,20 +29,62 @@ class Auth extends Component {
...
   }
 
   render () {
     const { loggedIn, kid, hasBackedUpKeys, loading } = this.props
 
-    if (loggedIn && !loading && kid) {
+    // here there is an extra condition, to show the webview only if the logged kid is the one we want to show
+    if (loggedIn && !loading && kid && this.state.selectedKid && this.state.selectedKid.id === kid) {
       return <Game />
     }
```
Comment (madclem) [inline src/navigation/auth/index.js]: extra condition, to only show the Game component (containing the webview) if the selected kid is the logged in kid!

File: src/navigation/auth/index.js [hunk 4]
```diff
@@ -57,13 +104,14 @@ class Auth extends Component {
       )
     }
 
-    return <Home />
+    return <Home screenProps={{ onKidSelect: this.onKidSelect }} />
   }
```
Comment (madclem) [inline src/navigation/auth/index.js]: add props to know if the user from the home clicks on a kid (means that he want to connect as a kid)

File: src/screens/home-kids/index.js [hunk 5]
```diff
@@ -17,6 +17,10 @@ class HomeKids extends Component {
     const hasValidPasscode = passcodeHash !== 'pending'
     const nextScreenId = hasValidPasscode ? SCREEN_KID_LOGIN : SCREEN_KID_SET_LOGIN
     this.props.navigation.navigate(nextScreenId, { kid })
+
+    // screen props when a kid is selected, passed by auth/index (then in navigation/home)
+    const onKidSelect = this.props.screenProps && this.props.screenProps.onKidSelect ? this.props.screenProps.onKidSelect : null
+    if (onKidSelect) onKidSelect(kid)
   }
```
Comment (madclem) [inline src/screens/home-kids/index.js]: just double check if the screen props is here, and call onKidSelect for `auth/index` to do its logic

File: src/screens/kid-login/index.js [hunk 6]
```diff
@@ -14,6 +14,16 @@ export class KidLogin extends Component {
       input: new Set()
     }
 
+    componentDidMount () {
+      this.backHandler = BackHandler.addEventListener('hardwareBackPress', () => {
```
Comment (ianmcgregor) [inline src/screens/kid-login/index.js]:
```suggestion:0+2
      this.backHandler = BackHandler.addEventListener('hardwareBackPress', this.onBack)
```

Comment (ianmcgregor) [reply src/screens/kid-login/index.js]: Don't need the extra anonymous function there

**Comment by madclem** [comment]
@ianmcgregor Not anymore, the backbutton signal only comes from the login screen now, so it should be safe! Good catch

**Pull request merged**